### PR TITLE
Solve Microsoft.AspNetCore.Http.Abstractions package deprecation

### PR DESCRIPTION
**Pull request by marcominerva**
## Motivation and Context (Why the change? What's the scenario?)

The Core project contains a reference to the Microsoft.AspNetCore.Http.Abstractions package: https://github.com/microsoft/kernel-memory/blob/main/service/Core/Core.csproj#L30. As we can read at https://github.com/dotnet/core/blob/main/release-notes/net-package-deprecation.md, this package has been deprecated.

See https://github.com/microsoft/kernel-memory/issues/342

## High level description (Approach, Design)

HTTP features are used only in the [HttpDocumentUploadRequest.cs](HttpDocumentUploadRequest.cs) and [HttpDocumentUploadRequestExtensions.cs](https://github.com/microsoft/kernel-memory/blob/main/service/Core/WebService/HttpDocumentUploadRequestExtensions.cs) files in the [WebService](https://github.com/microsoft/kernel-memory/tree/main/service/Core/WebService) folder of the **Core** project, so these have been moved to to Service.AspNetCore project, solving the deprecation issue.

### CODE REVIEW

**Comment by dluc** [comment]
Does this introduce another nuget package? Is it worth the complexity, just to remove a warning? About the deprecated package, why is that a problem?

**Comment by dluc** [comment]
The code doesn't build in dev mode. Here's the steps:

cp -p KernelMemory.sln KernelMemoryDev.sln
dotnet clean --nologo -v q -c Release KernelMemory.sln
dotnet clean --nologo -v q -c Debug KernelMemory.sln
dotnet clean --nologo -v q -c Release KernelMemoryDev.sln
dotnet clean --nologo -v q -c Debug KernelMemoryDev.sln
dotnet build --nologo -v m -c Debug KernelMemoryDev.sln

**Comment by marcominerva** [comment]
The compilation issue was related to https://github.com/microsoft/kernel-memory/issues/410. Now it is fixed.

File: service/WebService/WebService.csproj [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <ItemGroup>
+        <InternalsVisibleTo Include="Microsoft.Core.UnitTests" />
+    </ItemGroup>
+
+    <Import Project="../../code-analysis.props" />
+
+    <Import Project="../../nuget-package.props" />
+
+    <PropertyGroup>
+        <IsPackable>true</IsPackable>
```
Comment (dluc) [inline service/WebService/WebService.csproj:30]: why a new nuget package, what about just moving `HttpDocumentUploadRequest` and `HttpDocumentUploadRequestExtensions` to the Service project?

Comment (marcominerva) [reply service/WebService/WebService.csproj:30]: If it is OK to provide the features exposed by `HttpDocumentUploadRequest` and `HttpDocumentUploadRequestExtensions` only in the **Service** project, yes, this is a valid solution.

In this case, I can update the PR. Let me know 😀

**Review by dluc — APPROVED** [review]

**Comment by dluc** [comment]
GitHub is failing to merge the PR, support ticket created.

**Pull request merged**